### ISSUE TITLE
Add ability to predict on any model in isolation

### ISSUE THREAD

**Issue by btcross26** [btcross26/genestboost#20]
Issue: When calling predict, a user must either run the whole ensemble or the ensemble to a given index. However, when manual boosting with a for loop, this is inefficient.

Enhancement: Add the ability to predict on any model/learning rate combo in isolation - particularly, the last model. This would allow manual boosting with a holdout set without having to recalculate the whole model on the validation set every time one wanted validation set predictions. I.e., I could just track the holdout error and increment it at each for loop iteration by predicting on only the last model.
